### PR TITLE
add trace log in openbmc

### PR DESCRIPTION
**Pull request by xuweibj**
#3796 
xcatdebugmode=1
```
# rpower openbmctest stat
openbmctest: [openbmc_debug] curl -k -c cjar -H "Content-Type: application/json" -d '{ "data": ["root", "xxxxxx"] }' https://9.3.185.29/login
openbmctest: [openbmc_debug] login_response 200 OK
openbmctest: [openbmc_debug] curl -k -b cjar -X GET -H "Content-Type: application/json" https://9.3.185.29/xyz/openbmc_project/state/enumerate
openbmctest: [openbmc_debug] rpower_status_response 200 OK
openbmctest: off
```
in /var/log/xcat/cluster.log
```
Aug 30 03:02:56 c910f03c05k04 xcat[4289]: DEBUG [openbmc_debug] openbmctest curl -k -c cjar -H "Content-Type: application/json" -d '{ "data": ["root", "xxxxxx"] }' https://9.3.185.29/login
Aug 30 03:02:56 c910f03c05k04 xcat[4289]: DEBUG [openbmc_debug] openbmctest login_response 200 OK
Aug 30 03:02:56 c910f03c05k04 xcat[4289]: DEBUG [openbmc_debug] openbmctest curl -k -b cjar -X GET -H "Content-Type: application/json" https://9.3.185.29/xyz/openbmc_project/state/enumerate
Aug 30 03:02:57 c910f03c05k04 xcat[4289]: DEBUG [openbmc_debug] openbmctest rpower_status_response 200 OK
```

### CODE REVIEW

**Comment by gurevichmark** [comment]
I thought the idea for #3796 was to print to the console and the log what data comes back in the response. So for example `curl software/enumerate` would put the Dumper() style response objects into the log file.

File: xCAT-server/lib/xcat/plugins/openbmc.pm [hunk 1]
```diff
@@ -1067,19 +1068,20 @@ sub gen_send_request {
...
-        }
-        else {
-            $debug_info = "$method $request_url -d $content";
+    if ($xcatdebugmode) {
+        my $debug_info;
+        if ($method eq "GET") {
+            $debug_info = "curl -k -b cjar -X $method -H \"Content-Type: application/json\" $request_url";
+        } else {
+            if ($::UPLOAD_FILE) {
+                # Slightly different debug message when doing a file upload
```
Comment (whowutwut) [inline xCAT-server/lib/xcat/plugins/openbmc.pm]: @gurevichmark can you take a look here?  On the upload, would we want to print the command that we used to upload, for debug?

File: xCAT-server/lib/xcat/plugins/openbmc.pm [hunk 2]
```diff
@@ -1215,6 +1239,9 @@ sub rpower_response {
     xCAT_monitoring::monitorctrl::setNodeStatusAttributes(\%new_status, 1) if (%new_status);
 
     if ($node_info{$node}{cur_status} eq "RPOWER_STATUS_RESPONSE" and !$next_status{ $node_info{$node}{cur_status} }) { 
+
+        print Dumper(\%{ $response_info->{data} }) . "\n";
```
Comment (whowutwut) [inline xCAT-server/lib/xcat/plugins/openbmc.pm]: What's the purpose of adding the dumper out put here? And does it impact the performance of the commands? @robin2008 Is there any way to tell?   It's not wrapped in debugmode, so on all calls, we would be running this, regardless of bypass or not....  

And the output would only come out in non-hierarchy environments, we cannot use XCATBYPASS on hierarchy mode....

File: xCAT-server/lib/xcat/plugins/openbmc.pm (hunk 1 above)
Comment (gurevichmark) [reply xCAT-server/lib/xcat/plugins/openbmc.pm]: @whowutwut You are right.
@xuweibj The $request_url for the file upload on the next line should be:
`$request_url  = "$http_protocol://" . $node_info{$node}{bmc} . "/upload/image/"`

**Comment by whowutwut** [comment]
@gurevichmark I'm not sure we want to hold that much debug information in the log files.  

Here's the scenario that I'm trying to help with debugging... 
When customer reports some incorrect data in xCAT output on commands that gather data from the BMC.  We need to know what the BMC is sending back to us to verify whether it's a processing error on xCAT or the BMC data is incorrect. 

Debug would be: 
- As the user to rerun the command with `xcatdebugmode=1` in the site table

The problem here is that it only shows the request and not the response.  Then to ask the user to provide us with the response, we have to spend time to give them a correctly formatted curl command, Even for developers to come up with the command, it takes some time.  So to reduce this time spent formatting a correct CURL command, I suggested that we just print this out with the debug, so the code generates the command for the user to run, saving everyone time. 


Then the next question is whether we should only print this for debugmode=2... ?

File: xCAT-server/lib/xcat/plugins/openbmc.pm (hunk 2 above)
Comment (xuweibj) [reply xCAT-server/lib/xcat/plugins/openbmc.pm]: Yes, it's just for ourself to debug when using XCATBYPASS=1 in non-hierarchy environments. I think if there is some changes for ``rinv``, we could use it to debug quickly.

File: xCAT-server/lib/xcat/plugins/openbmc.pm (hunk 1 above)
Comment (xuweibj) [reply xCAT-server/lib/xcat/plugins/openbmc.pm]: @gurevichmark , sorry I did not get your point. Is the url defined below is wrong? If not, I think the url on the next line is OK. Or if it is wrong, we should modify the hash. On line 1065, we have added http and bmc in the string.
```
RFLASH_FILE_UPLOAD_REQUEST  => {
        method         => "PUT",
        init_url       => "$openbmc_project_url/upload/image/",
    },
```

File: xCAT-server/lib/xcat/plugins/openbmc.pm (hunk 2 above)
Comment (robin2008) [reply xCAT-server/lib/xcat/plugins/openbmc.pm]: We can use profile tool to see if the Dumper take more time,  but I think it should be less performance impact unless it is very large.
So if it is only for debugging,  we need check the environment XCATBYPASS or foreground mode before running it. Then it won't impact the customer and be convenient for our debugging.

File: xCAT-server/lib/xcat/plugins/openbmc.pm (hunk 1 above)
Comment (gurevichmark) [reply xCAT-server/lib/xcat/plugins/openbmc.pm]: @xuweibj It is a little hard to explain. Lets just merge this pull request as is, and I will do another pull request on top if it later to display correct info for "file upload"

**Pull request merged**